### PR TITLE
Complete navigation links and restrictions for all user types

### PR DESCRIPTION
**Pull request by grwthomps**
- Add appropriate links in navbar based on logged in user type 
- Restrict access to specific pages based on logged in user type

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -10,4 +10,12 @@ def cart
...
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
+
+  def current_merchant?
+    current_user && (current_user.merchant_employee? || current_user.merchant_admin?)
+  end
+
+  def current_admin?
+    current_user && current_user.admin?
+  end
```
Comment (johnktravers) [inline app/controllers/application_controller.rb:20]: Good catch adding question marks to the boolean methods

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -29,4 +31,10 @@ def increment_decrement
     end
     redirect_to "/cart"
   end
+
+  private
+
+  def disallow_admin
+    render file: '/public/404' if current_admin?
+  end
```
Comment (johnktravers) [inline app/controllers/cart_controller.rb:39]: Love the method name. Very formal

File: db/seeds.rb [hunk 3]
```diff
@@ -19,3 +20,8 @@
...
 dog_bone = dog_shop.items.create(name: "Dog Bone", description: "They'll love it!", price: 21, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
+
+default_user = User.create(name: "Lynda Ferguson", address: "452 Cherry St", city: "Tucson", state: "AZ", zip: 85736, email: "lferguson@gmail.com", password: "password1", password_confirmation: "password1")
+merchant_employee = User.create(name: "Lael Whipple", address: "7392 Oklahoma Ave", city: "Nashville", state: "TN", zip: 37966, email: "whip_whipple@yahoo.com", password: "password12", password_confirmation: "password12", role: 1)
+merchant_admin = User.create(name: "Dudley Laughlin", address: "2348 Willow Dr", city: "Big Sky", state: "MT", zip: 59716, email: "bigskyguy@aol.com", password: "password123", password_confirmation: "password123", role: 2)
+admin = User.create(name: "Dorian Bouchard", address: "7890 Montreal Blvd", city: "New Orleans", state: "LA", zip: 70032, email: "ouibouchard@gmail.fr", password: "password1234", password_confirmation: "password1234", role:3)
```
Comment (zacisaacson) [inline db/seeds.rb:27]: Great seed files

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -26,15 +26,37 @@
...
+            </li>
+          <% end %>
+          <% if current_admin? %>
+            <li class="nav-item">
+              <a class="nav-link" href="/admin">Dashboard</a>
+            </li>
+            <li class="nav-item">
+              <a class="nav-link" href="/admin/users">Users</a>
+            </li>
+          <% end %>
```
Comment (johnktravers) [inline app/views/layouts/application.html.erb:51]: Might want to look into using partials to slim down this view

File: db/seeds.rb (hunk 3 above)
Comment (johnktravers) [inline db/seeds.rb:27]: Excellent names! Very creative.

**Review by zacisaacson — APPROVED** [review]

File: spec/features/default_user/navigation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    user = User.create!(name: "Gmoney", address: "123 Lincoln St", city: "Denver", state: "CO", zip: 23840, email: "test@gmail.com", password: "password123", password_confirmation: "password123")
+
+    visit '/login'
+
+    fill_in :email, with: 'test@gmail.com'
+    fill_in :password, with: 'password123'
+
+    click_button 'Login'
+
+    expect(page).to have_content("Logged in as Gmoney")
```
Comment (johnktravers) [inline spec/features/default_user/navigation_spec.rb:14]: Let's do some more stubbing instead of logging in every time

**Review by johnktravers — COMMENTED** [review]
Great work! Everything looks good

**Pull request merged**